### PR TITLE
docs: update azure docs and fix descriptions

### PR DESCRIPTION
**Pull request by ianaya89**
## Affected Components
* [ ] Resources
* [ ] Data Sources
* [ ] Test
* [x] Docs
* [ ] Tooling
* [ ] Other

## Pre-Requisites
* [ ] Terraform and Go code are formatted with `make fmt`
* [ ] Documentation updated and re-generated using `make docs`

## Notes for the Reviewer

> Resolves #[issue-number]

### CODE REVIEW

File: docs/data-sources/env_azure_status.md [hunk 1]
```diff
@@ -51,4 +51,4 @@ Read-Only:
 
 Read-Only:
 
-- `private_link_service_alias` (String) VPC endpoint service name in $endpoint_service_id.$region.vpce.amazonaws.com format (if any)
+- `private_link_service_alias` (String) Globally unique name for your service in prefix.GUID.suffix format..
```
Comment (shyiko) [inline docs/data-sources/env_azure_status.md:54]: I suggest we change this to `Private Link Service Alias / DNS Name`. `... in prefix.GUID.suffix format` is really confusing.

File: docs/resources/env_azure.md [hunk 2]
```diff
@@ -10,7 +9,68 @@ description: |-
...
+
+variable "subscription_id" {
+  type        = string
+  description = "The ID of subscription to connect to Altinity.Cloud"
+}
+
+variable "tenant_id" {
+  type        = string
+  description = "The ID of tenant to connect to Altinity.Cloud"
+}
```
Comment (shyiko) [inline docs/resources/env_azure.md:29]: none of the other examples we have use variables. I suggest we remove these just to be consistent.

File: docs/resources/env_azure.md [hunk 3]
```diff
@@ -10,7 +9,68 @@ description: |-
...
+resource "azurerm_role_assignment" "altinity_cloud" {
+  name                 = random_uuid.azurerm_role_assignment_altinity_cloud.id
+  scope                = "/subscriptions/${var.subscription_id}"
+  role_definition_name = "Owner"
+  principal_id         = data.azuread_service_principal.altinity_cloud.object_id
+}
+
+locals {
+  region = "eastus"
+  zones  = ["eastus-1", "eastus-2"]
```
Comment (shyiko) [inline docs/resources/env_azure.md:49]: these case be inlined similar to other examples we have

File: docs/resources/env_azure.md [hunk 4]
```diff
@@ -10,7 +9,68 @@ description: |-
...
+    }
+  }
+
+  node_groups = [{
+    node_type         = "Standard_B2s_v2"
+    capacity_per_zone = 3
+    reservations      = ["CLICKHOUSE", "ZOOKEEPER", "SYSTEM"]
+  }]
+}
+```
```
Comment (shyiko) [inline docs/resources/env_azure.md:73]: we need to provide example showing how to provision env intended to be connected via PLS similar to https://github.com/Altinity/terraform-provider-altinitycloud/blob/master/examples/resources/altinitycloud_env_aws/vpce/main.tf

Comment (ianaya89) [reply docs/resources/env_azure.md:64]: I will work on this on a separate PR

**Pull request merged**